mlflow/models/evaluation/base.py

```diff
@@ -0,0 +1,530 @@
+            elif isinstance(self.data, pd.DataFrame):
+                EvaluationDataset._gen_md5_for_arraylike_obj(md5_gen, self.data)
+                md5_gen.update(self.labels.encode("UTF-8"))
+            self._hash = md5_gen.hexdigest()
+        return self._hash
+
+    @property
+    def _metadata(self):
+        """
+        Return dataset metadata containing name, hash, and optional path.
+        """
+        metadata = {
+            "name": self.name,
+            "hash": self.hash,
+        }
+        if self.path is not None:
+            metadata["path"] = self.path
+        return metadata
+
+    def _log_dataset_tag(self, client, run_id):
+        """
+        Log dataset metadata as a tag "mlflow.datasets", if the tag already exists, it will
+        append current dataset metadata into existing tag content.
+        """
+        existing_dataset_metadata_str = client.get_run(run_id).data.tags.get("mlflow.datasets")
+        if existing_dataset_metadata_str is not None:
+            dataset_metadata_list = json.loads(existing_dataset_metadata_str)
+        else:
+            dataset_metadata_list = []
+
+        metadata_exists = False
+        for metadata in dataset_metadata_list:
+            if metadata["hash"] == self.hash and metadata["name"] == self._user_specified_name:
+                metadata_exists = True
+                break
+
+        if not metadata_exists:
+            dataset_metadata_list.append(self._metadata)
+
+        dataset_metadata_str = json.dumps(dataset_metadata_list)
+        client.log_batch(
+            run_id,
+            tags=[RunTag("mlflow.datasets", dataset_metadata_str)],
+        )
+
+
+class ModelEvaluator:
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs) -> bool:
+        """
+        :param model_type: A string describing the model type (e.g., "regressor",
+                           "classifier", …).
+        :param evaluator_config: A dictionary of additional configurations for
+                                 the evaluator.
+        :param **kwargs: For forwards compatibility, a placeholder for additional
+                         arguments that may be added to the evaluation interface
+                         in the future.
+        :return: True if the evaluator can evaluate the specified model on the
+                 specified dataset. False otherwise.
+        """
+        raise NotImplementedError()
+
+    def _log_metrics(self, run_id, metrics, dataset_name):
+        """
+        Helper method to log metrics into specified run.
+        """
+        client = mlflow.tracking.MlflowClient()
+        timestamp = int(time.time() * 1000)
+        client.log_batch(
+            run_id,
+            metrics=[
+                Metric(key=f"{key}_on_{dataset_name}", value=value, timestamp=timestamp, step=0)
+                for key, value in metrics.items()
+            ],
+        )
```

Comment:
I don't think it's necessary to provide this method for now. Implementers can leverage `MlflowClient.log_batch()`.